### PR TITLE
Add address book upload analytics

### PR DESCRIPTION
**Pull request by daehn**
**What's in this PR?**

* Adds address book upload tracking ([6980](https://wearezeta.atlassian.net/browse/ZIOS-6980))

### CODE REVIEW

File: Source/Registration/ZMAddressBookEncoder.m [hunk 1]
```diff
@@ -167,6 +168,7 @@ - (void)createPayloadWithCompletionHandler:(void(^)(ZMEncodedAddressBook *encode
                     index++;
                 }
                 result.otherData = (otherHashesCards.count < 1) ? nil : otherHashesCards;
+                result.addressBookSize = addressBook.numberOfContacts;
```
Comment (marcoconti83) [inline Source/Registration/ZMAddressBookEncoder.m]: We still have that artificial limit of 1000 contacts max, right? In this case, I think `numberOfContacts` will be capped at 1000. What we wanted to know is whether the users has 12.000 contacts, so we can figure out if we are too restrictive with a limit of 1000.

File: Source/AddressBookTracker.swift [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import Foundation
+
+
+private enum Attribute: String {
+    case Outcome = "outcome"
+    case Size = "size"
+    case Interval = "interval"
+}
+
+private extension Bool {
```
Comment (mikeger) [inline Source/AddressBookTracker.swift]: I think that this is a bit weird. I would think of moving it into the scope of `AddressBookTracker`

File: Source/Registration/ZMAddressBookEncoder.m (hunk 1 above)
Comment (daehn) [inline Source/Registration/ZMAddressBookEncoder.m]: @marcoconti83 `numberOfContacts` was newly added and returns the number of all contacts in the address book

File: Source/AddressBookTracker.swift (hunk 2 above)
Comment (daehn) [inline Source/AddressBookTracker.swift]: @mikeger unfortunately extensions can only by on file scope, but as it is only used once I will inline it

**Comment by mikeger** [comment]
**All checks have failed**
Looks like Travis cannot find `ZMSystem` WAT

**Comment by daehn** [comment]
@mikeger travis is not yet fully set up for `wire-iso-sync-engine`

**Comment by mikeger** [comment]
@daehn All right, I see now

**Comment by daehn** [comment]
@marcoconti83 please re-review

**Pull request merged**